src/hotspot/os/posix/vmError_posix.cpp

```diff
@@ -49,16 +49,6 @@
 #endif
 
 
-// handle all synchronous program error signals which may happen during error
-// reporting. They must be unblocked, caught, handled.
-
-static const int SIGNALS[] = { SIGSEGV, SIGBUS, SIGILL, SIGFPE, SIGTRAP }; // add more if needed
-static const int NUM_SIGNALS = sizeof(SIGNALS) / sizeof(int);
-
-// Space for our "saved" signal flags and handlers
-static int resettedSigflags[NUM_SIGNALS];
-static address resettedSighandler[NUM_SIGNALS];
```

Comment:
I'm not  clear what happened to these ?? (I'm not clear how they were used in the first place. :( )

Reply:
I think they were used for path (3). When printing signal handlers, we do this little test to check - similar to what we do in Xcheck:jni - to print out if someone changed the handler under us. See:
https://github.com/openjdk/jdk/blob/cb127a4bb5d95d19eb1f5e625b600311a2490135/src/hotspot/os/posix/signals_posix.cpp#L1372
and
https://github.com/openjdk/jdk/blob/cb127a4bb5d95d19eb1f5e625b600311a2490135/src/hotspot/os/posix/signals_posix.cpp#L1389
.
But when printing signal handlers as part of a hs-err file, we replaced the hotspot signal handlers with the secondary crash handler and would trigger this test. I think this whole logic is needed just to prevent this.